obs_maven/repo.py

```diff
@@ -27,20 +27,31 @@
 
 
 class Repo:
-    def __init__(self, uri, project, repository):
-        self.uri = uri
-        self.project = project.replace(":", ":/")
-        self.repository = repository
+    def __init__(self, base_url, project, repository, custom_url=None):
+        self.base_url = base_url
+        self.custom_url = custom_url
+        if custom_url is None:
+            if project is not None and repository is not None:
+                self.project = project.replace(":", ":/")
+                self.repository = repository
+            else:
+                raise ValueError("Either 'project' and 'repository' or 'url' must be defined for the repository")
         self._rpms = None
 
+    def get_repo_path(self, path):
+        if self.custom_url is not None:
+            return "{}/{}".format(self.custom_url, path)
+        else:
+            return "{}/{}/{}/{}".format(self.base_url, self.project, self.repository, path)
```

Comment:
```python
import os
...
return os.path.joion(self.base_url), self.project, self.repository, path)
```

Reply:
it's not an OS path but rather a URL path

Reply:
Well, yes, the code says so. But you are still joining. If you don't want that, you can then simply:

```python
return "/".join([self.base_url, self.project, self.repository, path])
```